microsoft-integration/impl/src/main/java/org/sakaiproject/microsoft/impl/MicrosoftCommonServiceImpl.java

```diff
@@ -163,6 +167,10 @@ public class MicrosoftCommonServiceImpl implements MicrosoftCommonService {
 	final private Object lock = new Object();
 
 	public void init() {
+		LoggerContext context = (LoggerContext) LogManager.getContext(false);
+		context.setConfigLocation(new File(serverConfigurationService.getSakaiHomePath() + "custom-microsoft-log4j2.xml").toURI());
+		context.reconfigure();
+
```

Comment:
```suggestion
		log.info("Before Microsoft log reconfiguration");
		LoggerContext context = (LoggerContext) LogManager.getContext(false);
		context.setConfigLocation(new File(serverConfigurationService.getSakaiHomePath() + "custom-microsoft-log4j2.xml").toURI());
		context.reconfigure();
		log.info("After Microsoft log reconfiguration");
```

Reply:
He afegit aquest canvi i ara surt al catalina 
10-de juny-2024 12:55:47.336 ^[[0mINFO^[[m [main] o.s.m.i.MicrosoftCommonServiceImpl.init Before Microsoft log reconfiguration
i després escriu al fitxer de log 
[INFO ] 10-06-2024 12:55:47.372 [main] MicrosoftCommonServiceImpl - After Microsoft log reconfiguration
Però després si vaig a l'eina de sincro de Microsoft ja no escriu res al fitxer de log

Reply:
Pots executar el job per a vore si també apareix al fitxer de log de Microsoft?

Reply:
És estrany, pel que dius està llegint el xml i en el seu contingut està que mostre els logs de debug. De totes maneres, una solució provisional tenint en compte que el que es vol és redirigir el log a un fitxer diferent podria ser modificar els log.debug a log.info.

(Si els missatges del job també apareixen, té per començar i acabar)

Reply:
He executat el job i tampoc ha escrit al fitxer 
En canvi al catalina ha escrit logs d'error 
ERROR com.microsoft.aad.msal4j.ConfidentialClientApplication - [Correlation ID: 79c135cd-da75-49f0-aceb-a8f45ddf75bd] Execution of class com.microsoft.aad.msal4j.AcquireTokenByClientCredentialSupplier failed.
com.microsoft.aad.msal4j.MsalClientException: java.net.SocketTimeoutException: Read timed out
ERROR org.sakaiproject.microsoft.provider.AdminAuthProvider - Exception retrieving token from Microsoft Graph Auth Provider

Reply:
Se us acudeix alguna altra prova que pugui fer amb la configuració?

Reply:
`[DEBUG] 10-06-2024 13:37:25.865 [pool-8-thread-5] MicrosoftSynchronizationServiceImpl - Message listen from MicrosoftSynchronizationServiceImpl: TOPIC=ADD_MEMBER_TO_AUTHZGROUP => action=ADD, type=SITE, siteId=~admin, groupId=null, userId=admin, status=0, owner=true, forced=false
[DEBUG] 10-06-2024 13:37:25.868 [pool-8-thread-5] MicrosoftSynchronizationServiceImpl - NOT allowed to add user to team
[INFO ] 10-06-2024 13:37:27.247 [QuartzScheduler_Worker-2] RunSynchronizationsJob - RunSynchronizationsJob started.
[DEBUG] 10-06-2024 13:37:27.253 [pool-8-thread-6] MicrosoftSynchronizationServiceImpl - Message listen from MicrosoftSynchronizationServiceImpl: TOPIC=ADD_MEMBER_TO_AUTHZGROUP => action=ADD, type=SITE, siteId=~admin, groupId=null, userId=admin, status=0, owner=true, forced=false
[INFO ] 10-06-2024 13:37:27.253 [QuartzScheduler_Worker-2] RunSynchronizationsJob - RunSynchronizationsJob completed.
[DEBUG] 10-06-2024 13:37:27.254 [pool-8-thread-6] MicrosoftSynchronizationServiceImpl - NOT allowed to add user to team`

A nosaltres ens apareix tot redirigit al fitxer de Microsoft, amb compilar l'eina i reiniciar hauria de ser suficient. Potser alguna configuració del teu entorn pot estar afectant? Si ens dius podem intentar configurar-nos nosaltres el que siga diferent, no estem segurs del motiu

Reply:
De tota manera jo tornaria a posar org.sakaiproject.microsoft (sense el impl) si ho has canviat al xml i provaria el job d'eixa manera.